cytominer_database/tableSchema.py

```diff
@@ -0,0 +1,203 @@
+import os
+import csv
+import click
+import warnings
+import zlib
+import pandas as pd
+import backports.tempfile
+import sqlalchemy.exc
+from sqlalchemy import create_engine
+import pyarrow
+import pyarrow.parquet as pq
+import pyarrow.csv
+import numpy as np
+import collections
+import numpy as np
+import cytominer_database
+import cytominer_database.utils
+import cytominer_database.load
+
+################################################################################
+# Contains "open_writers()"" to generate the dictionary containing the
+# reference data ("writer dictionary"), from which the table schemata will be generated.
+# The reference tables can either be loaded directly from a designated folder,
+# or sampled across all available data, as specified in the config_file.
+#
+# Also contains helper functions "get_table_paths()" to generate a list of paths
+# and the sampling function get_reference_paths().
+################################################################################
+
+def open_writers(source, target, config_file, skip_image_prefix=True):
+    """
+    Determines, loads reference tables and openes them as ParquetWriters.
+    Returns a dictionary referencing the writers.
+    :param source: path to directory containing all parent folders of .csv files
+    :target: output file path
+    :config_file: parsed configuration data (output from cytominer_database.utils.read_config(config_path))
+    :skip_image_prefix: Boolean value specifying if the column headers of
+     the image.csv files should be prefixed with the table name ("Image").
+    :writers_dict: dictionary referencing the writers (return argument)
+    """
+    print("------------- in open_writers(): --------------")
+    if config_file["ingestion_engine"]["engine"] == "SQLite": # no reference table needed
+        return None
+
+    reference_directories = get_path_dictionary(config_file, source) #includes different steps, depending on config_file 
+    writers_dict = {} 
+    refIdentifier = 999 & 0xFFFFFFFF
+    # arbitrary identifier, will not be stored but used only as type template. (uint32 as in checksum())
+    # Iterate over all table kinds:
+    for name, path in reference_directories.items():  # iterates over keys of the dictionary reference_directories
+        # unpack path from [path] # if isinstance(path, list):
+        path = path[0]
```

Comment:
this seems a bit sketchy to me. Is path always going to be a list?

I see in the line 51 comment you say it might not be? why not then:

```python
if isinstance(path, list):
  path = path[0]
```

will this work?